### PR TITLE
Create data dictionary for current index findability subset

### PR DESCRIPTION
**Pull request by KutnerBroadie**
Added descriptions and required information for all columns and tables present in the current AnVIL Indexing schema.

### CODE REVIEW

**Comment by rsc3** [comment]
'target' is "A reference to the target recognized by the antibody."

File: anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json [hunk 1]
```diff
@@ -0,0 +1,888 @@
...
+    },
+    {
+      "name": "anvil_assayactivity",
+      /// not sure this makes sense
+      "description": "Activity table designed to capture information about assays related to files and biosamples.",
+      "columns": [
+        {
+          "name": "assayactivity_id", 
+          /// ?
+          "description": "Unique ID used to identify the assay instance represented by this record.",
```
Comment (ncalvanese1) [inline anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json]: Change "assay instance" to "assay activity instance"

File: anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json [hunk 2]
```diff
@@ -0,0 +1,888 @@
...
+    },
+    {
+      "name": "anvil_alignmentactivity",
+      /// ???
+      "description": "Activity table designed to capture information about the alignment process in next generation sequencing.",
+      "columns": [
+        {
+          "name": "alignmentactivity_id", 
+          /// ????
+          "description": "Unique ID used to identify the alignment instance represented by this record.",
```
Comment (ncalvanese1) [inline anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json]: Change "alignment instance" to "alignment activity instance"

File: anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json [hunk 3]
```diff
@@ -0,0 +1,888 @@
...
+          "name": "principal_investigator", 
+          "description": "The name of the principal investigator for this dataset.",
+          "datatype": "string", 
+          "array_of": true, 
+          "required": false
+        },
+        {
+          "name": "registered_identifier", 
+          /// probably needs a different description
+          "description": "Unique ID used to identify this dataset in an external database.",
```
Comment (ncalvanese1) [inline anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json:346]: For example, the Accession Number for dbGaP registered studies.

File: anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json [hunk 4]
```diff
@@ -0,0 +1,888 @@
...
+          "name": "file_ref", 
+          "description": "The fully qualified path to the file.",
+          "datatype": "fileref", 
+          "array_of": false, 
+          "required": false
+        },
+        {
+          "name": "is_supplementary", 
+          /// this could use a better definition, but isn't necessarily worse than some of the others....
+          "description": "Whether a file is considered supplementary or not.",
```
Comment (ncalvanese1) [inline anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json]: Indicates whether a file is considered supplementary or not, where supplementary is defined as a file that cannot be directly or indirectly tied back to a biosample record.

File: anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json [hunk 5]
```diff
@@ -0,0 +1,888 @@
...
+          "name": "title", 
+          "description": "Human-readable property naming or describing the project.",
+          "datatype": "string", 
+          "array_of": false, 
+          "required": false
+        },
+        {
+          "name": "registered_identifier", 
+          /// ??? Which one, from where? etc.
+          "description": "",
```
Comment (ncalvanese1) [inline anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json]: For example, the Accession Number for dbGaP registered studies.

File: anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json (hunk 3 above)
Comment (ncalvanese1) [inline anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json]: For example the DUOS Identifier for datasets registered in DUOS.

File: anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json [hunk 6]
```diff
@@ -0,0 +1,888 @@
+{
+  "version": 1,
+  "tables": [
+    {
+      "name": "anvil_activity",
+      /// ???
+      "description": "",
```
Comment (ncalvanese1) [inline anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json]: A table describing the activities performed on biosamples or files that resulted in the generation of additional files. The anvil_activity table is used to capture activity provenance that does not fit into the other activity tables described in the schema (such as file indexing, file checksums, etc.).

File: anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json [hunk 7]
```diff
@@ -0,0 +1,888 @@
...
+      "primaryKey": ["activity_id"],
+      "partitionMode": "none",
+      "datePartitionOptions": null,
+      "intPartitionOptions": null,
+      "reqiured": false
+    },
+    {
+      "name": "anvil_alignmentactivity",
+      /// ???
+      "description": "Activity table designed to capture information about the alignment process in next generation sequencing.",
```
Comment (ncalvanese1) [inline anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json]: Strike "next generation"

File: anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json (hunk 2 above)
Comment (ncalvanese1) [reply anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json]: Extrapolate to other unique ID instances on activity tables.

File: anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json [hunk 8]
```diff
@@ -0,0 +1,888 @@
...
+          "name": "antibody_id", 
+          "description": "Unique ID used to identify the antibody represented in this record."
+          "datatype": "string", 
+          "array_of": false, 
+          "required": true
+        },
+        {
+          "name": "target", 
+          // couldn't find a definition - all references say "definition needed"
+          "description": ""
```
Comment (ncalvanese1) [inline anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json]: See Sid's comment (in conversation)

File: anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json [hunk 9]
```diff
@@ -0,0 +1,888 @@
...
+      "primaryKey": ["antibody_id"],
+      "partitionMode": "none",
+      "datePartitionOptions": null,
+      "intPartitionOptions": null,
+      "reqiured": false
+    },
+    {
+      "name": "anvil_assayactivity",
+      /// not sure this makes sense
+      "description": "Activity table designed to capture information about assays related to files and biosamples.",
```
Comment (ncalvanese1) [inline anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json]: "Experimental measurement" --> Doesn't create files/biosamples, creates values/tabular data

File: anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json [hunk 10]
```diff
@@ -0,0 +1,888 @@
...
+      ],
+      "primaryKey": ["alignmentactivity_id"],
+      "partitionMode": "none",
+      "datePartitionOptions": null,
+      "intPartitionOptions": null,
+      "reqiured": false
+    },
+    {
+      "name": "anvil_antibody",
+      "description": "May be a reference to an antibody used for a SampleTreatmentActivity or another Activity.",
```
Comment (ncalvanese1) [inline anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json]: "SampleTreatmentActivity" --> "sample treatment activity"

"Activity" --> "activity"

"May be a reference..." --> "A reference to an..."

File: anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json [hunk 11]
```diff
@@ -0,0 +1,888 @@
...
+          "name": "onset_age_upper_bound", 
+          "description": "Upper bound for age of donor at documented time of onset. If exact age is known, this should be the same as the lower bound.",
+          "datatype": "float", 
+          "array_of": false, 
+          "required": false
+        },
+        {
+          "name": "phenotype", 
+          /// no idea what to put here wrt how this applies to the table
+          "description": "",
```
Comment (rsc3) [inline anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json]: A human-readable property that describes a disease, condition, or phenotype that has been reported in this entity in a human readable manner (as opposed to a code/id/etc). If multiple values exist, they must be synonymous.

File: anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json (hunk 9 above)
Comment (KutnerBroadie) [reply anvil_schema/fss_index_data_dictionary/fss_index_data_dictionary.json]: Couldn't remember how to implement this feedback - can you take a look and see if I got it right?

**Pull request merged**